src/layers.jl

```diff
@@ -127,18 +189,32 @@ end
 @functor Patching
 
 """
-    mlpblock(planes, expansion_factor = 4, dropout = 0., dense = Dense)
+    PosEmbedding{T}
 
-Feedforward block used in many vision transformer-like models.
+Positional embedding layer used by many vision transformer-like models. Instantiated with an 
+embedding vector which is a learnable parameter.
+"""
+struct PosEmbedding{T}
+  embedding_vector::T
+end
+
+(p::PosEmbedding)(x) = x .+ p.embedding_vector[:, 1:size(x)[2], :]
```

Comment:
Not in fact a vector. Can the field just be `array` or something? No need to repeat the struct's name.

Can the struct also be `PositionEmbedding` or something? "Pos" isn't saving much.

Reply:
It's a vector from the perspective of the paper, but in the practical code. Just like the patches are "vectors."

Maybe the name should be more specific since this type of positional embedding applies strictly to vision transformers? Perhaps `ViPostionEmbeddding` or `VisualPositionEmbedding` (that might be too long). The field can be called `signal` since we traditionally say "positional embedding signal."

Reply:
True, `PositionEmbedding` might be  too general.

Reply:
Could be like `.patchvectors` or something, if this helps match the paper. Even `p.vectors` seems fine. Or `signal` as you say.

Reply:
Transformers.jl just calls the field `embedding`: https://github.com/chengchingwen/Transformers.jl/blob/a646572c5e05ab0e752df678e101bc0fbb7d92b3/src/basic/embeds/position_embed.jl#L8-L11

Reply:
> It's a vector from the perspective of the paper, but not in the practical code. Just like the patches are "vectors."
> 
> Maybe the name should be more specific since this type of positional embedding applies strictly to vision transformers? Perhaps `ViPostionEmbeddding` or `VisualPositionEmbedding` (that might be too long). The field can be called `signal` since we traditionally say "positional embedding signal."

I've made this `ViPosEmbedding`...which while adding some confusion should be fine simply because it's highly unlikely that anyone but devs will ever need to use this layer. Plus docstring is pretty clear about its functionality